packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -0,0 +1,240 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math' as math;
+
+import 'package:flutter/animation.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+
+import 'scroll_metrics.dart';
+
+const double _kMinThumbExtent = 18.0;
+const Duration _kDefaultThumbFadeDuration = const Duration(milliseconds: 300);
+const Duration _kDefaultTimeToFade = const Duration(milliseconds: 600);
+
+/// A [CustomPainter] for painting scrollbars.
+///
+/// See also:
+///
+///  * [Scrollbar] for a widget showing a scrollbar around a [Scrollable] in the
+///    Material Design style.
+///  * [CupertinoScrollbar] for a widget showing a scrollbar around a
+///    [Scrollable] in the iOS style.
+class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
```

Comment:
IMHO we really shouldn't make the original _ScrollbarPainter class public. It's a crazy class with crazy semantics (e.g. it's a custom painter but you have to dispose it).

Reply:
If I left 2 implementations in 2 separate files, it seems to alleviate your concern here and on the next comment. Do you prefer that approach? A bit copy paste-y I initially thought.